src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java

```diff
@@ -0,0 +1,437 @@
+    public static final int                                          DEFAULT_META = BlockMinecoloniesRack.EnumType.DEFAULT.getMetadata();
+    public static final int                                          FULL_META    = EnumType.FULL.getMetadata();
+
+    /**
+     * The position it faces.
+     */
+    public static final PropertyDirection FACING = BlockHorizontal.FACING;
+
+    /**
+     * The hardness this block has.
+     */
+    private static final float BLOCK_HARDNESS = 10.0F;
+
+    /**
+     * This blocks name.
+     */
+    private static final String BLOCK_NAME = "blockMinecoloniesRack";
+
+    /**
+     * The resistance this block has.
+     */
+    private static final float RESISTANCE = Float.POSITIVE_INFINITY;
+
+    /**
+     * How much light goes through the block.
+     */
+    private static final int LIGHT_OPACITY = 0;
+
+    public BlockMinecoloniesRack()
+    {
+        super(Material.WOOD);
+        initBlock();
+    }
+
+    /**
+     * initialize the block
+     * sets the creative tab, as well as the resistance and the hardness.
+     */
+    private void initBlock()
+    {
+        setRegistryName(BLOCK_NAME);
+        setUnlocalizedName(String.format("%s.%s", Constants.MOD_ID.toLowerCase(), BLOCK_NAME));
+        setCreativeTab(ModCreativeTabs.MINECOLONIES);
+        GameRegistry.register(this);
+        GameRegistry.register((new ItemBlock(this)).setRegistryName(this.getRegistryName()));
+        this.setDefaultState(this.blockState.getBaseState().withProperty(FACING, EnumFacing.NORTH).withProperty(VARIANT, EnumType.DEFAULT));
+        setHardness(BLOCK_HARDNESS);
+        setResistance(RESISTANCE);
+        setLightOpacity(LIGHT_OPACITY);
+    }
+
+    @Override
+    public void getSubBlocks(final Item itemIn, final CreativeTabs tab, final List<ItemStack> list)
+    {
+        list.add(new ItemStack(this, 1, EnumType.DEFAULT.getMetadata()));
+    }
+
+    @Override
+    public void onBlockPlacedBy(
+            final World worldIn, final BlockPos pos, final IBlockState state, final EntityLivingBase placer, final ItemStack stack)
+    {
+        IBlockState tempState = state;
+        tempState = tempState.withProperty(VARIANT, EnumType.byMetadata(stack.getItemDamage()));
+        tempState = tempState.withProperty(FACING, placer.getHorizontalFacing().getOpposite());
+
+        worldIn.setBlockState(pos, tempState, 2);
+    }
+
+    //todo fix getActualState
+    /**
+     * @deprecated but we still need this because there is nothing better.
+     */
+    @Deprecated
+    @Override
+    public IBlockState getStateFromMeta(final int meta)
+    {
+        final EnumFacing enumFacing = EnumFacing.getHorizontal(meta);
+        return this.getDefaultState().withProperty(FACING, enumFacing);
+    }
+
+    /**
+     * @deprecated but we still need this because there is nothing better.
+     */
+    @Override
+    public IBlockState getActualState(final IBlockState state, final IBlockAccess worldIn, final BlockPos pos)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 7, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @Deprecated annotation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)